src/terms/pairwise.jl

```diff
@@ -0,0 +1,138 @@
+@doc raw"""
+Pairwise terms: Pairwise potential between nuclei, e.g., Van der Waals potentials, such as
+Lennard—Jones terms.
+The potential is dependent on the distance between to atomic positions and the pairwise
+atomic types:
+For a distance `d` between to atoms `A` and `B`, the potential is `V(d, params[(A, B)])`.
+The energy is of the form
+```math
+    ℰ = ∑_{1⩽i<j⩽N} V_{i,j} \left( \frac{1}{|r_i - r_j|} \right).
+```
+The parameters `max_radius` is of `100` by default, and gives the maximum distance between
+nuclei for which we consider interactions.
+"""
+struct PairwisePotential
+    V
+    params
+    max_radius
+end
+function PairwisePotential(V, params; max_radius=100)
+    PairwisePotential(V, params, max_radius)
+end
+(P::PairwisePotential)(basis::AbstractBasis) = TermPairwisePotential(P.V, P.params, P.max_radius, basis)
+
+# Even if the double loop is not performance critical, it is better to explicitely type
+# `max_radius` with `T` and not `Real`.
+struct TermPairwisePotential{TV, Tparams, T} <:Term
+    V::TV
+    params::Tparams
+    max_radius::T
+    energy::Real
+end
+function TermPairwisePotential(V, params, max_radius, basis::PlaneWaveBasis{T}) where {T}
+    TermPairwisePotential(V, params, max_radius, T(energy_pairwise(basis.model, V, params, max_radius)))
```

Comment:
don't cast it (eg if you're differentiating wrt the function V, energy_pairwise will return duals but T might not